content/blog/2019-11-odbc-1-2-0/index.Rmarkdown

```diff
@@ -0,0 +1,108 @@
+---
+title: odbc 1.2.0
+author: Jim Hester
+date: '2019-11-27'
+slug: odbc-1-2-0
+categories:
+  - package
+tags:
+  - odbc
+photo:
+  url: https://unsplash.com/photos/40XgDxBfYXM
+  author: Photo by Jordan Harrison
+---
+
+odbc 1.2.0 is now on CRAN! The odbc package provides a DataBase Interface (DBI) to the Open DataBase Connectivity (ODBC) drivers. ODBC drivers exist for nearly all widely used databases, including [SQL
+Server](https://www.microsoft.com/en-us/sql-server/),
+[Oracle](https://www.oracle.com/database), [MySQL](https://www.mysql.com/),
+[PostgreSQL](https://www.postgresql.org/), [SQLite](https://sqlite.org/) and
+others. The odbc package will allow you to query these databases from within R. Get the latest version with:
+
+```r
+install.packages("odbc")
+```
+
+The biggest improvements in this release are related to querying within schemas, an API for immediate execution, and a new `timezone_out` parmeter to control the displayed time of timezones returned by the query.
+
+## Querying schemas
+
+Use `DBI::Id()` reference a table within a schema. This will allow odbc to handle any quoting necessary for your particular database. e.g. you can write to a table named `my.iris` even if there is a schema named `my`.
+
+```{r, eval = FALSE}
+library(DBI)
+
+con <- dbConnect(odbc::odbc(), "MicrosoftSQLServer", UID = "SA", PWD = "Password12")
+
+my_tbl <- Id(schema = "my", table = "my.iris")
+dbWriteTable(con, my_tbl, iris)
+
+tbl <- dbReadTable(con, my_tbl)
+# The quoting for "my"."my.iris" is done automatically when 
+# using DBI::Id()
+tbl2 <- dbGetQuery(con, 'SELECT * FROM "my"."my.iris"')
+all.equal(tbl, tbl2)
+#> TRUE
+```
+
+This feature has actually existed for a number of odbc releases, but due to driver inconsistencies was not working properly on SQL Server, which is now fixed.
+
+## Immediate/Direct execution
+
+The odbc package uses [Prepared Statements](https://en.wikipedia.org/wiki/Prepared_statement), to compile the query once and reuse it, allowing large or repeated queries to be more efficient. However prepared statements can actually perform worse in some cases, such as many different small queries that are all only executed once. Because of this the odbc package now also supports direct queries by specifying `immediate = TRUE`.
```

Comment:
```suggestion
The odbc package uses [Prepared Statements](https://en.wikipedia.org/wiki/Prepared_statement) to compile the query once and reuse it, allowing large or repeated queries to be more efficient. However, prepared statements can actually perform worse in some cases, such as many different small queries that are all only executed once. Because of this the odbc package now also supports direct queries by specifying `immediate = TRUE`.
```